lib/data/submission.js

```diff
@@ -71,8 +71,16 @@ const submissionXmlToFieldStream = (fields, xml, includeStructuralAttrs = false,
     }
   }, { xmlMode: true, decodeEntities: true });
 
-  parser.write(xml);
-  parser.end();
+  if (!xml) {
+    outStream.destroy(new Error('Stream ended before stack was exhausted.'));
+  } else {
+    if (xml instanceof Buffer) {
+      parser.write(xml.toString('utf8'));
```

Comment:
Did you consider streaming buffers into the parser?  It feels like it would be more efficient than converting the buffer to an intermediate string.

Reply:
Oh I see!